packages/block-library/src/navigation/index.php

```diff
@@ -350,6 +350,40 @@ function block_core_navigation_get_fallback_blocks() {
 	return apply_filters( 'block_core_navigation_render_fallback', $fallback_blocks );
 }
 
+/**
+ * Iterate through all inner blocks recursively and get navigation link block's post ids..
+ *
+ * @param WP_Block_List $inner_blocks Block list class instance.
+ *
+ * @return array Array of post ids.
+ */
+function block_core_navigation_get_post_ids( $inner_blocks ){
+	$post_list = array_map( 'get_post_id_from_navigation_blocks', iterator_to_array( $inner_blocks ) );
+	return array_unique( array_merge( ...$post_list ) );
```

Comment:
The `array_merge` here shouldn't be necessary?
```suggestion
	return array_unique( $post_list );
```

Reply:
No, this is very much needed to flatten the array. Without this, it wold be an array of arrays.